File: plugins/gumshoe/base/backend/types/skills.py
from __future__ import annotations
from typing import Any, Protocol, Literal, Optional, runtime_checkable
from pydantic import BaseModel, Field, NonNegativeInt, ConfigDict


SkillKind = Literal["investigative", "general"]

class SkillGroup(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: str
    title: str
    color: str

    kind: SkillKind


class Skill(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: str
    title: str
    group: str

File: plugins/gumshoe/base/ui/src/types/skills.ts
export type SkillGroup = {
  id: string;
  title: string;
  color: string;
  kind: "investigative" | "general";
};

export type Skill = {
  id: string;
  title: string;
  group: string; // "investigative_*" | "general_*"
};

export type SkillsConfig = {
  skillGroups: SkillGroup[];
  skills: Skill[];
};
